server/pkg/accounts/accounts.go

```diff
@@ -3,14 +3,39 @@ package accounts
 import (
 	"fmt"
 	"os"
+	"errors"
 
 	"github.com/darwinfroese/hacksite/server/models"
 	"github.com/darwinfroese/hacksite/server/pkg/auth"
 	"github.com/darwinfroese/hacksite/server/pkg/database"
 )
 
+const (
+	// UnathorizedErrorMessage is used to determine if the error is because of unathorized account
+	UnathorizedErrorMessage = "Account is invalid"
+)
+
 // CreateAccount will create an account and insert it into the database
 func CreateAccount(db database.Database, account *models.Account) error {
+
+	//Check if the username already exists
+	_, invalidUsername := db.GetAccount(account.Username)
+	if invalidUsername != nil {
+		//Username is not in use
+	} else {
+		invalidUsername = errors.New("Username is taken")
+		return invalidUsername
+	}
+
+	//Check if the email already exists
+	_, invalidEmail := db.GetAccountByEmail(account.Email)
+	if invalidEmail != nil {
+		//Email is not in use
+	} else {
+		invalidEmail = errors.New("This Email is already in use")
```

Comment:
errors is a fine package, but i prefer `fmt.Errorf("...", ...)` since you can use formatting and output additional information alongside the error that you're echoing if it exists.